cmd/daemon/main.go

```diff
@@ -93,12 +94,31 @@ func (app *App) newAppPlayer(creds any) (_ *AppPlayer, err error) {
 	if appPlayer.player, err = player.NewPlayer(
 		appPlayer.sess.Spclient(), appPlayer.sess.AudioKey(),
 		!app.cfg.NormalisationDisabled, *app.cfg.NormalisationPregain,
-		appPlayer.countryCode, *app.cfg.AudioDevice, *app.cfg.MixerDevice,
-		*app.cfg.VolumeSteps, app.cfg.ExternalVolume,
+		appPlayer.countryCode, *app.cfg.AudioDevice, *app.cfg.MixerDevice, *app.cfg.MixerControlName,
+		*app.cfg.VolumeSteps, app.cfg.ExternalVolume, appPlayer.externalVolumeUpdate,
 	); err != nil {
 		return nil, fmt.Errorf("failed initializing player: %w", err)
 	}
 
+	// only update the "spotify volume", when external volume is enabled
+	// try to keep synchronized with the device volume
+	if app.cfg.ExternalVolume {
+		// listen on external volume changes (for example the alsa driver)
+		go func() {
+			for {
+				v, ok := <-appPlayer.externalVolumeUpdate
+				if !ok {
+					break
+				}
+
+				appPlayer.updateVolume(uint32(v * player.MaxStateVolume))
+
+				// prevent "too many requests"
+				time.Sleep(2 * time.Second)
```

Comment:
I am not entirely convinced by this because volume change events will just pile up in the queue and be delayed by so many seconds, I think this can be removed.

If ALSA is very spammy about this we may want to throttle instead by discarding all events that are too near to each another.

Reply:
The problem here is, that everytime the alsa volume changes, another request gets fired, so if you have for example something like a phone and just hold the volume down, alsa sends multiple requests per second which leads to too many requests errors. 

This is also the reason while I use a queue in this way, because every time a new volume update gets send, the queue gets emptied before, so nothing can pile up.

Reply:
That is what I imagined, the rewrite with the `RingBuffer` looks much better because the logic can be reused easily.